### DISCUSSION TITLE
Moving kafka to a new cluster

### DISCUSSION THREAD

**Discussion by alok87** [strimzi/strimzi-kafka-operator#4151] [Q&A] [answered]
We have a kafka on the cluster A
`kubectl get kafka --context=clusterA.example.com`
We want to take it to cluster B
`kubectl get kafka --context=clusterB.example.com`

our persistence volume is AWS EBS.
A small downtime of streaming is acceptable for us but without any data loss in Kafka.
We have a single kafka connect pod that writes to this kafka. We can stop it and do this activity.

Please suggest how can this be done, without doing a fresh setup(new EBS).

**Comment by scholzj** [comment, marked as answer]
I guess you have two options:
* You can move the storage. You can stop the old cluster and have the volumes unmounted. And then you can use them to create a new PVs in the cluster B. Once you have the PVs, you can create the PVCs with the right names. And then when you deploy Kafka in cluster B, it should reuse the PVCs with the PVs and the data on them. I'm not completely sure what does it take to move the PVs => partially because I don't know your setup (Are the Kube clusters in the same account? same region? Same AZs? Same VPC?). They might also need to be some changes you would need to do to the labels etc. to not get them deleted by accident later. So the safe option would be to shutdown the original cluster, snapshot the PVs and create new PVs from the volumes.
* Other option is to just use Mirror Maker and mirror the data.

**Reply by alok87** [reply]
Thanks for the answer.
